Cesium3DTilesSelection/src/ImplicitOctreeLoader.cpp

```diff
@@ -121,86 +121,94 @@ CesiumAsync::Future<TileLoadResult> requestTileContent(
-                                  pCompletedRequest) mutable {
-        const CesiumAsync::IAssetResponse* pResponse =
-            pCompletedRequest->response();
-        auto fail = [&]() {
-          return asyncSystem.createResolvedFuture(
-              TileLoadResult::createFailedResult(std::move(pCompletedRequest)));
-        };
-        const std::string& tileUrl = pCompletedRequest->url();
-        if (!pResponse) {
-          SPDLOG_LOGGER_ERROR(
-              pLogger,
-              "Did not receive a valid response for tile content {}",
-              tileUrl);
-          return fail();
-        }
-
-        uint16_t statusCode = pResponse->statusCode();
-        if (statusCode != 0 && (statusCode < 200 || statusCode >= 300)) {
-          SPDLOG_LOGGER_ERROR(
-              pLogger,
-              "Received status code {} for tile content {}",
-              statusCode,
-              tileUrl);
-          return fail();
-        }
-
-        // find gltf converter
-        const auto& responseData = pResponse->data();
-        auto converter = GltfConverters::getConverterByMagic(responseData);
-        if (!converter) {
-          converter = GltfConverters::getConverterByFileExtension(
-              pCompletedRequest->url());
-        }
-
-        if (converter) {
-          // Convert to gltf
-          CesiumGltfReader::GltfReaderOptions gltfOptions;
-          gltfOptions.ktx2TranscodeTargets = ktx2TranscodeTargets;
-          gltfOptions.applyTextureTransform = applyTextureTransform;
-          AssetFetcher assetFetcher{
-              asyncSystem,
-              pAssetAccessor,
-              tileUrl,
-              tileTransform,
-              requestHeaders,
-              CesiumGeometry::Axis::Y};
-          return converter(responseData, gltfOptions, assetFetcher)
-              .thenImmediately([pLogger, tileUrl, pCompletedRequest, ellipsoid](
-                                   GltfConverterResult&& result) {
-                // Report any errors if there are any
-                logTileLoadResult(pLogger, tileUrl, result.errors);
-                if (result.errors || !result.model) {
-                  return TileLoadResult::createFailedResult(
-                      std::move(pCompletedRequest));
-                }
-
-                return TileLoadResult{
-                    std::move(*result.model),
-                    CesiumGeometry::Axis::Y,
-                    std::nullopt,
-                    std::nullopt,
-                    std::nullopt,
-                    std::move(pCompletedRequest),
-                    {},
-                    TileLoadResultState::Success,
-                    ellipsoid};
-              });
-        }
-        // content type is not supported
-        return fail();
-      });
+      .thenInWorkerThread(
+          [pLogger,
+           ktx2TranscodeTargets,
+           applyTextureTransform,
+           &asyncSystem,
+           pAssetAccessor = pAssetAccessor,
```

Comment:
Oh yes, this annoyed me too. I almost feel like I'm missing something. But as far as I can tell, this is just how it works...

So we want to be able to std::move that pAssetAccessor, right? We can only do that if it's not const. That requires the lambda to be mutable, otherwise everything that's captured is const. But if the thing captured is const to begin with, it _also_ means that we need to do this funny assignment thing. Because otherwise, if we just listed `pAssetAccessor` in the capture, even though the lambda gets a _copy_ of it, the copy is still const. For some reason. 

Check out this godbolt:
https://godbolt.org/z/6WY7qs4E8

Notice the `s = s`. Change it to simply `s` and you'll get compiler errors because `s` is now const.

Reply:
Urk.